api/collections.js

```diff
@@ -74,20 +75,26 @@ function getPublicCollection(req, res, next) {
 }
 
 function listPublicCollectionQuotes(req, res, next) {
-  const query = { type: 'public', _id: req.params.id };
+  const query = { type: Types.Public, _id: req.params.id };
   QuoteCollection.findOne(query)
     .then(quoteCollection => {
       if (!quoteCollection) {
         return res.status(HTTPStatus.NO_CONTENT).end();
       }
       return quoteCollection.getQuotes()
-        .then(quotes => res.status(HTTPStatus.OK).send(quotes));
+        .then(quotes => {
+         Promise.all(quotes.map((quote) => {
+           if (quote.type === Types.Public) return quote;
+           else return quote.type;
+         }))
+          .then(publicQuotes => res.status(HTTPStatus.OK).send(publicQuotes));
+        });
```

Comment:
Something is not right here. What are you trying to achieve with this block of code? 😕

Reply:
Uh, since you have to ask, I guess it's no good :) Anyway let me try to explain. This route fetches and returns all quotes from the public quote collection. What I mean by public quote collections is a collection that user decided to be available to general public audience. But also we have to block quotes that are private, because only public quotes (what I mean by public is approved by admin through voting process) must be seen and rendered. This could be solved by not allowing the collection to go public if it contains any private quotes, but i prefer this way instead. No reason to block the whole collections to go public because of a few unapproved (private) quotes. By returning quote.type back when the quote is private, we have knowledge that more quotes exist in collection but they are not available to public yet.

Reply:
So all this can be _translated_ as `collection.getQuotes({ type: Types.Public })` :wink:

Reply:
ahahah that solution crossed my mind long before this whole "shabang" but you as you can remember I modified the getQuotes so it calls another method findManyById which has to return all the quotes because it's used not only for this purpose, but anyway I will try to implement one more query parameter to pass along or something... `  getQuotes() {
    const Quote = mongoose.model('Quote');
    return Quote.findManyById(this.quotes);
  },` 
Such a nice example how one optimization makes another more complex .... 😞